tests/log_parser/test_performance_parser.py

```diff
@@ -12,8 +12,11 @@ def test_performance_log_parsing_malformed_perfherder_data():
 
     # invalid json
     parser.parse_line("PERFHERDER_DATA: {oh noes i am not valid json}", 1)
-    # doesn't comply with schema
-    parser.parse_line("PERFHERDER_DATA: {}", 2)
```

Comment:
I'm not sure why we require a more explicit exception. All that matters is the fact that the passed JSON is wrong & doesn't conform to our schema. This old test proofs Perfherder is working as expected.

Reply:
The reason we need a more explicity error is because otherwise we could not get back to the block (see code below) that has information about the log's url:
```python 
 try:
     # Using `replace` to prevent malformed unicode (which might possibly exist
     # in test message output) from breaking parsing of the rest of the log.
     builder.parse_line(line.decode('utf-8', 'replace'))
 except EmptyPerformanceData:
     logger.warning("We have parsed an empty PERFHERDER_DATA for %s", self.url)
```

Reply:
I tried the code that would pass a 3rd parameter but it would get messy and odd because so many classes are sharing from the same base class.